xorsearch/xorsearch/xorsearch.py

```diff
@@ -52,11 +53,13 @@ async def scan(self, payload: Payload, request: Request) -> WorkerResponse:
         """
         result: Dict = {}
 
-        cmd = [self.bin, '-f', self.terms, '-']
-        p = Popen(cmd, stdout=PIPE, stdin=PIPE, stderr=PIPE)
-        out, err = p.communicate(input=payload.content)
-        process_results = out.splitlines()
-        for line in process_results:
+        with tempfile.NamedTemporaryFile() as temp_file:
```

Comment:
@mlaferrera just curious why you went back to creating tempfiles instead of reading from stdin?

Reply:
Sadly xorsearch was not accepting stdin with async PIPE. If you'd like to take a look at it and see why, please do.